packages/edit-post/src/plugins/index.js

```diff
@@ -15,21 +16,37 @@ import KeyboardShortcutsHelpMenuItem from './keyboard-shortcuts-help-menu-item';
 import ToolsMoreMenuGroup from '../components/header/tools-more-menu-group';
 import WelcomeGuideMenuItem from './welcome-guide-menu-item';
 
+function ManagePatternsMenuItem() {
+	const defaultUrl = addQueryArgs( 'edit.php', { postType: 'wp_block' } );
+	const patternsUrl = addQueryArgs( 'site-editor.php', {
+		path: '/patterns',
+	} );
+
+	const [ url, setUrl ] = useState( defaultUrl );
+
+	useEffect( () => {
+		window.fetch( patternsUrl ).then( ( response ) => {
```

Comment:
This is clever but I'm not a fan 😀

The problem is that the result isn't cached so the component will perform an AJAX request every time it mounts. If we use `canUser` then it will only perform the `OPTIONS` request once.

My preference would be to:
- Return `null` if user can’t access the Manage Patterns screen.
  - Determine this using `canUser( 'read', 'global-styles' )`
- Link to the site editor if `getSettings().__unstableIsBlockBasedTheme` is true